rfcs/0003-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,90 @@
+# Buildpack.yml to Environment Variables
+
+## Proposal
+
+Migrate to using environment variables to do all buildpack configuration and
+get rid of `buildpack.yml`.
+
+## Motivation
+
+There are several reasons for making this switch.
+1. There is already an existing RFC that proposes moving away from
+   `buildpack.yml` as a configuration tool.
+1. Environment variables appears to be the standard for configuration in other
+   buildpack ecosystems such as Google Buildpacks and Heroku as well as the
+   Paketo Java buildpacks. Making this change will align the buildpack with the
+   rest of the buildpack ecosystem.
+1. There is native support to pass environment variables to the buildpack
+   either on a per run basis or by configuration that can be checked into
+   source control, in the form of `project.toml`.
+1. Go and Dotnet-core buildpacks are moving to this strategy.
+
+## Implementation
+The proposed environment variables are as follows:
+
+### Existing configurations
+
+#### BP_NODE_VERSION
+```shell
+$BP_NODE_VERSION="~10"
+```
+This will replace the following structure in `buildpack.yml`:
+```yaml
+nodejs:
+  version: ~10
+```
+
+#### BP_NODE_OPTIMIZE_MEMORY
+```shell
+$BP_NODE_OPTIMIZE_MEMORY=true
+```
+This will replace the following structure in `buildpack.yml`:
+```yaml
+nodejs:
+  optimize-memory: true
+```
+
+### Proposed additions
+
+#### BP_NODE_PROJECT_PATH
+```shell
+$BP_NODE_PROJECT_PATH=./src/nodejs
+```
+If set, this relative path will be considered as the root of the app to be
+built by nodejs buildpacks.
+
+There is no current `buildpack.yml` support for this.
+
+#### BP_NODE_START_ENTRYPOINT
+```shell
+$BP_NODE_START_ENTRYPOINT=customserver.js
+```
+Only applicable when using the `node-start` buildpack. This sets the file to
+call as argument to node. When both `$BP_NODE_PROJECT_PATH` and
+`$BP_NODE_START_ENTRYPOINT` are set, the `node-start` buildpack will set the
+start command as as `node $BP_NODE_PROJECT_PATH/$BP_NODE_START_ENTRYPOINT`.
```

Comment:
Caveat: I don't frequently write node projects, however...

I think the `BP_NODE_START_ENTRYPOINT` env var being relative to the `BP_NODE_PROJECT_PATH` is a bit confusing/adds potential for odd configuration choices. Maybe a clunkier UX to set `BP_NODE_START_ENTRYPOINT=path/to/my/entrypoint.js`, but also makes sure buildpack uses are clear about what file they want run.

Reply:
I agree with @fgj. I think it is clearer if the relative base of the environment variable does not change based on the value of a different variable.